src/hotspot/share/gc/shared/weakProcessorPhases.hpp

```diff
@@ -40,44 +40,14 @@ class WeakProcessorPhases : AllStatic {
 
   typedef void (*Processor)(BoolObjectClosure*, OopClosure*);
 
-  enum Phase {
-    // Serial phase.
-    JVMTI_ONLY(jvmti)
-
-    // Additional implicit phase values follow for oopstorages.
-  };
-
-  static const uint serial_phase_start = 0;
-  static const uint serial_phase_count = 0 JVMTI_ONLY(+ 1);
-  static const uint oopstorage_phase_start = serial_phase_count;
+  static const uint oopstorage_phase_start = 0;
   static const uint oopstorage_phase_count = OopStorageSet::weak_count;
-  static const uint phase_count = serial_phase_count + oopstorage_phase_count;
-
-  // Precondition: value < serial_phase_count
-  static Phase serial_phase(uint value);
-
-  // Precondition: value < oopstorage_phase_count
-  static Phase oopstorage_phase(uint value);
-
-  // Indexes relative to the corresponding phase_start constant.
-  // Precondition: is_serial(phase) or is_oopstorage(phase) accordingly
-  static uint serial_index(Phase phase);
-  static uint oopstorage_index(Phase phase);
-
-  static bool is_serial(Phase phase);
-  static bool is_oopstorage(Phase phase);
+  static const uint phase_count = oopstorage_phase_count;
 
-  static Iterator serial_iterator();
   static Iterator oopstorage_iterator();
-
-  // Precondition: is_serial(phase)
-  static const char* description(Phase phase);
-
-  // Precondition: is_serial(phase)
-  static Processor processor(Phase phase);
 };
 
-typedef WeakProcessorPhases::Phase WeakProcessorPhase;
+typedef uint WeakProcessorPhase;
```

Comment:
Ok, so I'm not sure what to do with this:

  enum Phase { 
    // Serial phase.
    JVMTI_ONLY(jvmti)
  
    // Additional implicit phase values follow for oopstorages.
  };

I've removed the only thing in this enum.

Reply:
>Though it might be possible to go even further and eliminate WeakProcessorPhases as a thing separate from OopStorageSet.

This makes sense.  Can we file another RFE for this? I was sort of surprised by how much code was involved so I tried to find a place to stop deleting.

Reply:
I restored the empty enum for Phase and can open a new RFE if there is more code to remove.